src/functional/map.go

```diff
@@ -0,0 +1,9 @@
+package functional
+
+func Map[In any, Out any](arr []In, transformFn func(it In) Out) []Out {
+	ret := make([]Out, len(arr))
+	for i, it := range arr {
+		ret[i] = transformFn(it)
+	}
+	return ret
+}
```

Comment:
```suggestion
func Map[In any, Out any](array []In, transformFn func(input In) Out) []Out {
	mappedResult := make([]Out, len(array))
	
	for i, element := range array {
		mappedResult[i] = transformFn(element)
	}
	
	return mappedResult
}
```

Reply:
Somehow I f'ed up the intendation, please disregard the w i d e tabs :sweat_smile: